File: app/src/main/java/hanas/aptacy/moodlamp/services/pojo/LeadingColorBody.java
package hanas.aptacy.moodlamp.services.pojo;

import android.graphics.Color;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class LeadingColorBody {

    @SerializedName("color")
    @Expose
    private List<Integer> color = null;
    @SerializedName("pixel_count")
    @Expose
    private Integer pixelCount;

    public int getColor() {

        return  Color.rgb(color.get(0), color.get(1), color.get(2));
    }

    public float[] getHSV() {
        float[] hsv = new float[3];;
        Color.RGBToHSV(color.get(0), color.get(1), color.get(2), hsv);
        return hsv;
    }

    public void setColor(List<Integer> color) {
        this.color = color;
    }

    public Integer getPixelCount() {
        return pixelCount;
    }

    public void setPixelCount(Integer pixelCount) {
        this.pixelCount = pixelCount;
    }

}

File: app/src/main/java/hanas/aptacy/moodlamp/services/leadingcolors/pojo/LeadingColorBody.kt
package hanas.aptacy.moodlamp.services.leadingcolors.pojo

import android.graphics.Color
import com.google.gson.annotations.SerializedName

data class LeadingColorBody(
        @SerializedName("color") var color: List<Int>,
        @SerializedName("pixel_count") var pixelCount: Int
) {
    fun getColorInt(): Int {
        return Color.rgb(color[0], color[1], color[2])
    }

    val hsv: FloatArray
        get() {
            val hsv = FloatArray(3)
            Color.RGBToHSV(color[0], color[1], color[2], hsv)
            return hsv
        }
}